MAUI/DataForm/DataForm-settings.md

```diff
@@ -0,0 +1,853 @@
+    {
+        e.Cancel = true;
+    }
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+The `GenerateDataFormItem` event will be raised only when `AutoGenerateItems` is set to true.
+
+For `AutoGenerateItems` false you can use settings properties directly by using the `DataFormItem`. Please refer here to know more about creating data editors explicitly.
+
+## Change the auto-generated editor settings
+
+You can change the data editor settings by using the property of `DataFormItem` in the `GenerateDataFormItem` event when `AutoGenerateItems` is true.
+
+Here, `Salary` data field is restricted from being edited in the data form.
+
+{% tabs %}
+{% highlight C# %}
+
+this.dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+{
+    if (e.DataFormItem != null)
+    {
+        if (e.DataFormItem.FieldName == "MiddleName")
+        {
+             e.DataFormItem.IsReadOnly = true;
+        }
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+#### Change the autogenerated editor visibility
+
+You can change the specific editor visibility by using the `IsVisible` property in the `DataFormItem`.
+
+Here, `LastName` data field will be hidden.
+
+{% tabs %}
+{% highlight C# %}
+
+this.dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+{
+    if (e.DataFormItem != null)
+    {
+        if (e.DataFormItem.FieldName == "LastName")
+        {
+             e.DataFormItem.IsVisible = false;
+        }
+    }
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+
+## Setting watermark for auto-generated editor
+
+You can display the watermark in the editor by defining the display attribute or using the `GenerateDataFormItem` event when `AutoGenerateItems` is true.
+
+#### Using attribute
+
+You can show the watermark in the editor by setting the `Prompt` in display attribute.
+
+{% tabs %}
+{% highlight C# %}
+
+[Display(Prompt = "Enter middle name")]
+public string MiddleName { get; set; }
+    
+{% endhighlight %}
+{% endtabs %}
+
+#### Using event
+
+You can show the watermark in the editor by using the `PlaceholderText` property in `DataFormItem`.
+
+{% tabs %}
+{% highlight C# %}
+
+this.dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+{
+    if (e.DataFormItem != null)
+    {
+        if (e.DataFormItem.FieldName == "Description")
+        {
+            e.DataFormItem.PlaceholderText = "Enter description";
+        }
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+
+#### Changing watermark color for auto-generated editor
+
+You can display the color for the watermark in the editor by using `GenerateDataFormItem` event when `AutoGenerateItems` is true.
+
+You can change the color for the watermark in the editor by using the `PlaceholderColor` property in `DataFormItem`.
```

Comment:
You can change the color for the watermark in the editor by using the `PlaceholderColor` property in the `DataFormItem.`

Reply:
updated